model/postgres/postgres_resource.rb

```diff
@@ -59,6 +59,23 @@ def connection_string
     URI::Generic.build2(scheme: "postgres", userinfo: "postgres:#{URI.encode_uri_component(superuser_password)}", host: hostname).to_s if hostname
   end
 
+  def replication_connection_string(application_name:)
+    query_parameters = {
+      sslrootcert: "/dat/16/data/ca.crt",
+      sslcert: "/dat/16/data/server.crt",
+      sslkey: "/dat/16/data/server.key",
+      sslmode: "verify-full",
+      application_name: application_name
+    }.map { |k, v| "#{k}=#{v}" }.join("\\&")
+
+    URI::Generic.build2(scheme: "postgres", userinfo: "ubi_replication", host: identity, query: query_parameters).to_s
+  end
+
+  def required_standby_count
+    required_standby_count_map = {"none" => 0, "async" => 1, "sync" => 2}
```

Comment:
Is there some way we can use constants or something like that in the ruby code to avoid the kind of hazards incurred by freehanding strings like this? Or does it bloat the code?

Reply:
I added constants for these. It bloats a little, but OK price for not dealing with strings directly.